ariadne/src/Ariadne/Wallet/UiAdapter.hs

```diff
@@ -21,26 +32,36 @@ import Ariadne.Wallet.Cardano.Kernel.DB.InDb
 import Ariadne.Wallet.Cardano.Kernel.DB.Util.IxSet
 import Ariadne.Wallet.Face
 
-import Control.Lens (makeLenses)
+import Pos.Core (unsafeIntegerToCoin)
 import Serokell.Util (enumerate)
 
+import Control.Lens (makeLenses)
+
 -- 'WalletData' like data type, used only in UI glue
 data UiWalletData = UiWalletData
   { _uwdName     :: !Text
   , _uwdId       :: !HdRootId
   , _uwdAccounts :: !(Vector UiAccountData)
+  , _uwdBalance  :: !Coin
   } deriving (Show, Generic)
 
 data UiAccountData = UiAccountData
   { _uadName      :: !Text
   , _uadPath      :: !Word32
   -- Because of ChainType layer Now it should be ((HdAddressChain, Word32), Address) I guess, but
   -- AFAIU we don't want a new layer, so addresses of both types wiil be in one list -- External first.
-  , _uadAddresses :: !(Vector (Word32, Address))
+  , _uadAddresses :: !(Vector (Word32, UiAddressData))
```

Comment:
Is there a reason not to put `Word32` into `UiAddressData` to make types simpler?